### PR TITLE
sof: Define possible UUID entries usage in IPC during declaration

### PR DESCRIPTION
**Pull request by ktrzcinx**
A few UUID entries, related with components from topology,
should be loaded into SRAM to allow runtime comparison with
value in IPC message during component creation. UUID of the
components whose are not created by IPC message are not needed
in runtime, so there should not be there to reduce memory
consumtion.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

It's a part of https://github.com/thesofproject/sof/pull/2920
@keyonjie please check if I choose 'RT` UUID for all needed components.

### CODE REVIEW

**Comment by paulstelian97** [comment]
We'll wait until we're sure this won't cause any Out Of Memory issues in DSP SRAM.

**Comment by ktrzcinx** [comment]
From CI:
```The command "./scripts/docker-run.sh ./scripts/host-build-all.sh" exited with 0.
0.13s$ ./scripts/host-testbench.sh
==========================================================
test volume with ./volume_run.sh 16 16 48000 zeros_in.raw volume_out.raw
volume test failed!```

@singalsu maybe you have a suggestion what is going on ?

File: src/include/sof/lib/uuid.h [hunk 1]
```diff
@@ -84,17 +90,48 @@ struct sof_uuid_entry {
...
+#define DECLARE_SOF_RT_UUID(entity_name, uuid_name,			\
+			 va, vb, vc,					\
+			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7)	\
+	DECLARE_SOF_UUID(entity_name, uuid_name,			\
+			 va, vb, vc,					\
+			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7);	\
+	const struct sof_uuid uuid_name = {				\
+		.a = va, .b = vb, .c = vc,				\
+		.d = {vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7}		\
+	}
```
Comment (keyonjie) [inline src/include/sof/lib/uuid.h:128]: So both 'uuid_name ## ldc' and 'uuid_name' sof_uuid struct will be located in section .static_uuids with your change? This sounds odd to me. Do you need to change the linker scripts also?

Comment (ktrzcinx) [reply src/include/sof/lib/uuid.h:128]: no, only `uuid_name ## _ldc` will be in dedicated section, `uuid_name` should be in `.rodata` section

Comment (keyonjie) [reply src/include/sof/lib/uuid.h:128]: OK, then it makes sense, I don't need the linker script for my UUID PR any more after this applied.

Comment (ktrzcinx) [reply src/include/sof/lib/uuid.h:128]: Right, the whole content of yours first commit (_platform: move uuid_entries to RODATA part of the FW_) should be dropped and this PR  content should be used instead.

**Comment by singalsu** [comment]
> From CI:
> 
> ```
> 0.13s$ ./scripts/host-testbench.sh
> ==========================================================
> test volume with ./volume_run.sh 16 16 48000 zeros_in.raw volume_out.raw
> volume test failed!
> ```
> 
> @singalsu maybe you have a suggestion what is going on ?

OK, I'll see what  causes that.

**Comment by singalsu** [comment]
@ktrzcinx It segfaults to this line 

`0x00007ffff7e4e8bd in comp_new (comp=comp@entry=0x7fffffffd4a0) at /home/singalsu/work/current/sof/sof/src/audio/component.c:79`

`tr_info(&comp_tr, "comp new %s type %d id %d.%d", drv->tctx->uuid_p, comp->type, comp->pipeline_id, comp->id);
`
drv->tctx is NULL

Where is it set normally? Probably when updating this it has been forgotten to add to testbench runtime too. The trace output in testbench has been somewhat broken for a while already due to left  without updates but this is fist time that it segfaults.

**Comment by ktrzcinx** [comment]
@singalsu Thanks, `comp_file_host` and `comp_file_dai` didn't have assigned any pointer to `.tctx`, which one I start to use in `comp_new()`. So I added one more commit and now should be ok, I guess.

**Comment by singalsu** [comment]
> @singalsu Thanks, `comp_file_host` and `comp_file_dai` didn't have assigned any pointer to `.tctx`, which one I start to use in `comp_new()`. So I added one more commit and now should be ok, I guess.

Superb, thanks!! Yes I just ran the process_test and everything worked again.

**Comment by lgirdwood** [comment]
Jenkins DUT failure.

**Pull request merged**